Orange/widgets/unsupervised/owtsne.py

```diff
@@ -248,154 +431,159 @@ def set_data(self, data: Table):
             else:
                 self.normalize_cbx.setToolTip("")
 
-    def pca_preprocessing(self):
-        """Perform PCA preprocessing before passing off the data to t-SNE."""
-        if self.pca_data is not None:
-            return
-
-        projector = PCA(n_components=self.pca_components, random_state=0)
-        # If the normalization box is ticked, we'll add the `Normalize`
-        # preprocessor to PCA
-        if self.normalize:
-            projector.preprocessors += (preprocess.Normalize(),)
-
-        model = projector(self.data)
-        self.pca_data = model(self.data)
+        if self._invalidated:
+            self.run()
 
-    def __start(self):
-        self.pca_preprocessing()
+    def run(self):
+        self._set_modified(False)
 
-        self.needs_to_draw = True
-
-        # We call PCA through fastTSNE because it involves scaling. Instead of
-        # worrying about this ourselves, we'll let the library worry for us.
-        initialization = TSNE.default_initialization(
-            self.pca_data.X, n_components=2, random_state=0)
+        # When the data is invalid, it is set to `None` and an error is set,
+        # therefore it would be erroneous to clear the error here
+        if self.data is not None:
+            self.Error.clear()
```

Comment:
Is is possible to have any error messages show at this point?

Reply:
Yes, you're right. This remained here from earlier when we were catching general errors in t-SNE optimization. We later agreed that it was better to crash so the error could be reported and not hidden away in the red error box thing.